internal/fwserver/server_listresource.go

```diff
@@ -80,12 +79,17 @@ type ListResult struct {
 var NoListResults = func(func(ListResult) bool) {}
 
 // ListResource implements the framework server ListResource RPC.
-func (s *Server) ListResource(ctx context.Context, fwReq *ListRequest, fwStream *ListResultsStream) error {
+func (s *Server) ListResource(ctx context.Context, fwReq *ListRequest, fwStream *ListResultsStream) diag.Diagnostics {
 	listResource := fwReq.ListResource
```

Comment:
```suggestion
func (s *Server) ListResource(ctx context.Context, fwReq *ListRequest, fwStream *ListResultsStream) {
```

This seems consistent with adjacent methods that have no return value, e.g. `ReadResource`.

Instead, we can push a single `ListResult` with Diagnostics to `fwStream`.